bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/thing/ThingResource.java

```diff
@@ -522,16 +525,41 @@ public Response updateFirmware(
     public Response getFirmwareStatus(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) String language,
             @PathParam("thingUID") @ApiParam(value = "thing") String thingUID) throws IOException {
         ThingUID thingUIDObject = new ThingUID(thingUID);
+        FirmwareStatusDTO firmwareStatusDto = getThingFirmwareStatusInfo(thingUIDObject);
+        if (firmwareStatusDto == null) {
+            return Response.status(Status.NO_CONTENT).build();
+        }
+
+        return Response.ok(firmwareStatusDto, MediaType.APPLICATION_JSON).build();
+    }
 
-        FirmwareStatusInfo info = firmwareUpdateService.getFirmwareStatusInfo(thingUIDObject);
-        if (info == null) {
+    @GET
+    @Path("/{thingUID}/firmwares")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Get all available firmwares for provided thing UID", response = StrippedThingTypeDTO.class, responseContainer = "Set")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 204, message = "No firmwares found.") })
+    public Response getFirmwares(@PathParam("thingUID") @ApiParam(value = "thingUID") String thingUID,
+            @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(value = HttpHeaders.ACCEPT_LANGUAGE) String language) {
+        ThingUID aThingUID = new ThingUID(thingUID);
+        Thing thing = thingRegistry.get(aThingUID);
+        Collection<Firmware> firmwares = firmwareRegistry.getFirmwares(thing, LocaleUtil.getLocale(language));
```

Comment:
Don't you want to return a `404` in case`thing == null`? At least in the console command you're checking that and the `FirmwareRegistry` doesn't like `null`-Things...

Reply:
Yes, there is even a private method that I can reuse.

Reply:
And will also check the firmware version for null or empty as previously this check was done for the FirmwareUID.